### PR TITLE
[Merged by Bors] - chore(Topology/PartialHomeomorph): rename type variables

### PR DESCRIPTION
**Pull request by grunweg**
Greek letters are dead, long live X, Y and Z. Same procedure as in previous renames.

---

Note for reviewers
- ignore the first three commits; they are from a dependent PR which has now been merged
- the next six commits belong to this PR and can be reviewed individually
- the remaining three commits are just merging master; I'd squash them if I knew how...
- [x] depends on: #9630
- [x] depends on: #9633

### CODE REVIEW

File: Mathlib/Topology/PartialHomeomorph.lean [hunk 1]
```diff
@@ -1041,8 +1041,8 @@ section Prod
 /-- The product of two partial homeomorphisms, as a partial homeomorphism on the product space. -/
 @[simps! (config := mfld_cfg) toPartialEquiv apply,
   simps! (config := .lemmasOnly) source target symm_apply]
-def prod (e : PartialHomeomorph α β) (e' : PartialHomeomorph γ δ) :
-    PartialHomeomorph (α × γ) (β × δ) where
+def prod (e : PartialHomeomorph X Y) (e' : PartialHomeomorph Z Z') :
```
Comment (urkud) [inline Mathlib/Topology/PartialHomeomorph.lean:1044]: I would use something like `(eX : PartialHomeomorph X X') (eY : PartialHomeomorph Y Y')`, similarly to [Homeomorph.prodCongr](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Homeomorph.html#Homeomorph.prodCongr).

File: Mathlib/Topology/PartialHomeomorph.lean [hunk 2]
```diff
@@ -1051,37 +1051,37 @@ def prod (e : PartialHomeomorph α β) (e' : PartialHomeomorph γ δ) :
...
+    (PartialHomeomorph.refl X).prod (PartialHomeomorph.refl Y) = PartialHomeomorph.refl (X × Y) :=
   PartialHomeomorph.ext _ _ (fun _ => rfl) (fun _ => rfl) univ_prod_univ
 #align local_homeomorph.refl_prod_refl PartialHomeomorph.refl_prod_refl
 
 @[simp, mfld_simps]
-theorem prod_trans {η : Type*} {ε : Type*} [TopologicalSpace η] [TopologicalSpace ε]
-    (e : PartialHomeomorph α β) (f : PartialHomeomorph β γ) (e' : PartialHomeomorph δ η)
-    (f' : PartialHomeomorph η ε) : (e.prod e').trans (f.prod f') = (e.trans f).prod (e'.trans f') :=
+theorem prod_trans (e : PartialHomeomorph X Y) (f : PartialHomeomorph Y Z)
+    (e' : PartialHomeomorph Z' Z'') (f' : PartialHomeomorph Z'' Z''') :
```
Comment (urkud) [inline Mathlib/Topology/PartialHomeomorph.lean:1067]: ... and with `PartialHomeomorph Z Z'` here.

**Comment by urkud** [comment]
Otherwise LGTM. Thanks!
bors d+

File: Mathlib/Topology/PartialHomeomorph.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Topology/PartialHomeomorph.lean]: That's much better; switched to it.

File: Mathlib/Topology/PartialHomeomorph.lean (hunk 2 above)
Comment (grunweg) [reply Mathlib/Topology/PartialHomeomorph.lean]: I have used `PartialHomeomorph X Y`, `PartialHomeomorph Y Z` and then `PartialHomeomorph X' Y'` and `PartialHomeomorph Y' Z'`.

**Comment by grunweg** [comment]
Thank you for the review.
bors r+

**Pull request closed without merging**